src/Compilers/Core/Portable/InternalUtilities/ReferenceEqualityComparer.cs

```diff
@@ -10,25 +10,25 @@ namespace Roslyn.Utilities
     /// <summary>
     /// Compares objects based upon their reference identity.
     /// </summary>
-    internal class ReferenceEqualityComparer : IEqualityComparer<object>
+    internal class ReferenceEqualityComparer : IEqualityComparer<object?>
     {
         public static readonly ReferenceEqualityComparer Instance = new ReferenceEqualityComparer();
 
         private ReferenceEqualityComparer()
         {
         }
 
-        bool IEqualityComparer<object>.Equals(object? a, object? b)
+        bool IEqualityComparer<object?>.Equals(object? a, object? b)
         {
             return a == b;
         }
 
-        int IEqualityComparer<object>.GetHashCode(object a)
+        int IEqualityComparer<object?>.GetHashCode(object? a)
```

Comment:
IEqualityComparer.GetHashCode is annotated with `DisallowNull` in the framework, seems like this shouldn't accept null.

Reply:
I'm forced to put `?` here, but I could put `[DisallowNull]`. On the other hand, the implementation does handle `null`, so that's why I left it as-is.